internal/controller/accessrequest_controller.go

```diff
@@ -269,11 +269,9 @@ func buildResult(status api.Status, ar *api.AccessRequest, config config.Control
 		result.Requeue = true
 		result.RequeueAfter = time.Until(ar.Status.ExpiresAt.Time)
 	default:
-		if isConcluded(ar) && hasTTLConfig(config) {
-			if ttl := getTTLTime(ar, config); ttl != nil {
-				result.Requeue = true
-				result.RequeueAfter = time.Until(*ttl)
-			}
+		if ttl := getTTLTime(ar, config); ttl != nil && hasTTLConfig(config) {
```

Comment:
We still need the `isConcluded()` call as part of the `if` condition.

Reply:
To have this code clear I suggest:

1. refactor the `getTTLTime` function and make it return `time.Time`, not a pointer
2. remove the `isConcluded` call from the `getTTLTime` as that isn't the responsibility of that function.
3. `buildResult` if should be like the original code:
```
		if isConcluded(ar) && hasTTLConfig(config) {
			result.Requeue = true
			result.RequeueAfter = getTTLTime(ar, config).Sub(time.Now())
		}
```
WDYT?